dipy/workflows/reconst.py

```diff
@@ -861,3 +862,122 @@ def get_fitted_tensor(self, data, mask, bval, bvec, b0_threshold=50):
+
+        Parameters
+        ----------
+        input_files : string
+            Path to the input volumes. This path may contain wildcards to
+            process multiple inputs at once.
+        bvalues_files : string
+            Path to the bvalues files. This path may contain wildcards to use
+            multiple bvalues files at once.
+        bvectors_files : string
+            Path to the bvalues files. This path may contain wildcards to use
+            multiple bvalues files at once.
+        split_b_D : int, optional
+            Value to split the bvals to estimate D for the two-stage process of
+            fitting
+            (default 400)
+        split_b_S0 : int, optional
+            Value to split the bvals to estimate S0 for the two-stage process
+            of fitting
+            (default 200)
+        save_metrics : variable string, optional
+            List of metrics to save.
+            Possible values: S0_predicted, perfusion_fraction, D_star, D
+            (default [] (all))
+        out_dir : string, optional
+            Output directory (default input file directory)
+        out_S0_predicted : string, optional
+            Name of the S0 signal estimated to be saved
+            (default: 'S0_predicted.nii.gz')
+        out_perfusion_fraction : string, optional
+            Name of the estimated volume fractions to be saved
+            (default 'perfusion_fraction.nii.gz')
+        out_D_star : string, optional
+            Name of the estimated pseudo-diffusion parameter to be saved
+            (default 'D_star.nii.gz')
+        out_D : string, optional
+            Name of the estimated diffusion parameter to be saved
+            (default 'D.nii.gz')
+
+        References
+        ----------
+
+        .. [Stejskal65] Stejskal, E. O.; Tanner, J. E. (1 January 1965).
+                        "Spin Diffusion Measurements: Spin Echoes in the
+                        Presence of a Time-Dependent Field Gradient". The
+                        Journal of Chemical Physics 42 (1): 288.
+                        Bibcode: 1965JChPh..42..288S. doi:10.1063/1.1695690.
+
+        .. [LeBihan84] Le Bihan, Denis, et al. "Separation of diffusion
+                       and perfusion in intravoxel incoherent motion MR
+                       imaging." Radiology 168.2 (1988): 497-505.
+        """
+
+        io_it = self.get_io_iterator()
+
+        for (dwi, bval, bvec, oS0_predicted, operfusion_fraction,
+             oD_star, oD) in io_it:
+
+            logging.info('Computing IVIM metrics for {0}'.format(dwi))
+            data, affine = load_nifti(dwi)
+
+            ivimfit, _ = self.get_fitted_ivim(data, bval, bvec,
+                                              b0_threshold=0)
+
+            if not save_metrics:
+                save_metrics = ['S0_predicted', 'perfusion_fraction', 'D_star',
+                                'D']
+
+            S0_predicted = ivimfit.S0_predicted
```

Comment:
Do you really need to refer to these here? You could just refer directly to the `ivimfit` attributes below as needed.

Reply:
Will do as per your suggestion! Thanks :)